server/services/referralService.test.ts

```diff
@@ -112,6 +112,32 @@ describe('ReferralService', () => {
     })
   })
 
+  describe('getNumberOfTasksCompleted', () => {
+    let referralService: ReferralService
+
+    beforeEach(() => {
+      referralService = createMock<ReferralService>(service)
+    })
+
+    it.each`
+      referralKeys                                                                                                | expected
```

Comment:
This table test is really cool. I found it a bit tricky to read at first, I wonder if we could make it even easier to scan with variable names like `referralFormFields` and `expectedTasksCompleted` (something a bit more verbose)?